packages/host/app/components/card-catalog/filters.gts

```diff
@@ -1,19 +1,55 @@
 import Component from '@glimmer/component';
-import { IconButton } from '@cardstack/boxel-ui';
+import { BoxelDropdown, IconButton, Menu } from '@cardstack/boxel-ui';
+import { MenuItem } from '@cardstack/boxel-ui/helpers/menu-item';
+import { fn } from '@ember/helper';
 
 export default class CardCatalogFilters extends Component {
+  get realmMenuItems() {
+    return this.args.availableRealms.map((realm) => {
+      return new MenuItem(realm.name, 'action', {
+        action: () => {
+          let isSelected = this.args.selectedRealms
+            .map((r) => r.url)
+            .includes(realm.url);
+
+          // Toggle selection - if the item selected, deselect it, if not, select it
+          if (isSelected) {
+            // Prevent deselecting all realms. At least one realm must be selected at all times
+            if (this.args.selectedRealms.length > 1) {
+              this.args.onDeselectRealm(realm);
+            }
+          } else {
+            this.args.onSelectRealm(realm);
+          }
+
+          return false;
+        },
+        selected: this.args.selectedRealms
+          .map((r) => r.url)
+          .includes(realm.url),
+      });
+    });
+  }
+
+  get realmFilterSummary() {
+    return this.args.selectedRealms.map((realm) => realm.name).join(', ');
+  }
+
   <template>
     <div class='filters'>
-      <IconButton
-        class='add-filter-button'
-        @icon='icon-plus'
-        @width='20'
-        @height='20'
-        aria-label='add filter'
-      />
       <ul class='filter-list'>
         <li class='filter'>
-          Realm: All
+          <BoxelDropdown>
+            <:trigger as |bindings|>
+              <button class='pill' {{bindings}}>
```

Comment:
`pill` can be removed, it serves no purpose. It looks like a remnant from some test code